File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    return sum(range(1, number + 1)) ** 2

def sum_of_squares(number):
    numbers = list(range(1, number + 1))
    return sum(map(lambda value: value ** 2, numbers))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: ruby/difference-of-squares/difference_of_squares.rb
# frozen_string_literal: true

class Squares
  attr_reader :number

  def initialize(number)
    @number = number
  end

  def square_of_sum
    (1..number).sum**2
  end

  def sum_of_squares
    (1..number).sum(0) { |num| num**2 }
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
